src/main/webapp/user_profile_script.js

```diff
@@ -301,4 +308,277 @@ async function displayOfferHelpCompleteTasks() {
+                "featureType": "transit.line",
+                "elementType": "labels.text.stroke",
+                "stylers": [{"color": "#ebe3cd"}]
+            },
+            {
+                "featureType": "transit.station",
+                "elementType": "geometry",
+                "stylers": [{"color": "#dfd2ae"}]
+            },
+            {
+                "featureType": "water",
+                "elementType": "geometry.fill",
+                "stylers": [{"color": "#65d3f9"}, {"saturation": -10},  {"lightness": 10}]
+            },
+            {
+                "featureType": "water",
+                "elementType": "labels.text.fill",
+                "stylers": [{"color": "#92998d"}]
+            }
+        ],
+    });
+    map.setTilt(45);
+
+    var geocoder = new google.maps.Geocoder();
+    var infowindow = new google.maps.InfoWindow();
+
+    // When the map is clicked, display a marker and fill out the address info
+    map.addListener("click", function(event) {
+        var marker = displayMarker(event.latLng);
+        geocodeLatLng(geocoder, map, infowindow, event.latLng, marker);
+    });
+
+    var input = document.getElementById("place-input");
+    var searchBox = new google.maps.places.SearchBox(input);
+    map.controls[google.maps.ControlPosition.TOP_LEFT].push(input);
+
+    // Restrict the search result near to the current viewport
+    map.addListener("bounds_changed", function() {
+        searchBox.setBounds(map.getBounds());
+    });
+
+    // Place a new marker on the place that the user searches
+    searchBox.addListener("places_changed", function() {
+        var places = searchBox.getPlaces();
+        if (places.length === 0) {
+            return;
+        }
+
+        var bounds = new google.maps.LatLngBounds();
+        places.forEach(function(place) {
+            if (!place.geometry) {
+                return;
+            }
+            if (place.geometry.viewport) {
+                bounds.union(place.geometry.viewport);
+            } else {
+                bounds.extend(place.geometry.location);
+            }
+        });
+        map.fitBounds(bounds);
+    });
+
+    function onError() {
+        console.log("Unable to resolve the user's current location");
+    }
+
+    function onSuccess(geo) {
+        const position = {
+            lat: geo.coords.latitude,
+            lng: geo.coords.longitude
+        };
+
+        // Set the center to the user's current location
+        map.setCenter(position);
+    }
+
+    if (!navigator.geolocation) {
+        onError();
+    } else {
+        await navigator.geolocation.getCurrentPosition(onSuccess, onError);
+    }
+}
+
+function displayMarker(position) {
+    var lat = position.lat();
+    var lng = position.lng();
+    let marker = new google.maps.Marker({
+        position: {lat: lat, lng: lng},
+        map: map
+    });
+
+    google.maps.event.addListener(marker, "dblclick", function(event) {
+        deleteMarker(lat, lng);
```

Comment:
Should we also clear the address and zip code inputs when deleting a marker?

Reply:
Sure. I should definitely do that. In the new commit, when the user deletes a marker, the address, zipcode and country input will be set to empty and a placeholder would be shown to remind the user to fill out the blank.